src/components/Tool/FixEmailAddresses/FixEmailAddresses.tsx

```diff
@@ -288,7 +288,14 @@ export const FixEmailAddresses: React.FC<FixEmailAddressesProps> = ({
                     name={`${person.firstName} ${person.lastName}`}
                     key={person.id}
                     personId={person.id}
-                    emails={dataState[person.id]?.emailAddresses || []}
+                    emails={
+                      dataState[person.id]?.emailAddresses.map((email) => ({
+                        ...email,
+                        isValid: false,
+                        personId: person.id,
+                        isPrimary: email.primary,
+                      })) || []
+                    }
                     toDelete={dataState[person.id]?.toDelete}
```

Comment:
Remember to do this with useMemo rather than doing it in the component, so not to cause react to re-render a bunch.